x-pack/plugins/actions/server/builtin_action_types/torq.test.ts

```diff
@@ -0,0 +1,254 @@
+    expect(actionType.id).toEqual('.torq');
+    expect(actionType.name).toEqual('Torq');
+  });
+});
+
+describe('secrets validation', () => {
+  test('succeeds when secrets is valid', () => {
+    const secrets: Record<string, string> = {
+      token: 'jfi2fji3ofeaiw34if',
+    };
+    expect(validateSecrets(actionType, secrets)).toEqual(secrets);
+  });
+
+  test('fails when secret token is not provided', () => {
+    expect(() => {
+      validateSecrets(actionType, {});
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"error validating action type secrets: token is required"`
+    );
+  });
+});
+
+describe('config validation', () => {
+  const defaultValues: Record<string, string | null> = {};
+
+  test('config validation passes with an appropriate endpoint', () => {
+    const config: Record<string, string | boolean> = {
+      webhookIntegrationUrl: 'https://hooks.torq.io/v1/test',
+    };
+    expect(validateConfig(actionType, config)).toEqual({
+      ...defaultValues,
+      ...config,
+    });
+  });
+
+  ['iamnotavalidurl', 'example.com/do-something'].forEach((url) => {
+    test('should validate and throw error when an invalid URL is provided', () => {
+      const config: Record<string, string> = {
+        webhookIntegrationUrl: url,
+      };
+      expect(() => {
+        validateConfig(actionType, config);
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"error validating action type config: error configuring send to Torq action: unable to parse url: TypeError: Invalid URL: ${url}"`
+      );
+    });
+  });
+
+  test('fails when URL is not a Torq webhook endpoint', () => {
+    const config: Record<string, string> = {
+      webhookIntegrationUrl: 'http://mylisteningserver:9200/endpoint',
+    };
+    expect(() => {
+      validateConfig(actionType, config);
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"error validating action type config: error configuring send to Torq action: url must begin with https://hooks.torq.io"`
+    );
+  });
+
+  test("config validation returns an error if the specified URL isn't added to allowedHosts", () => {
+    actionType = getActionType({
+      logger: mockedLogger,
+      configurationUtilities: {
+        ...actionsConfigMock.create(),
+        ensureUriAllowed: (_) => {
+          throw new Error(`target url is not present in allowedHosts`);
+        },
+      },
+    });
+
+    // any for testing
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const config: Record<string, any> = {
+      webhookIntegrationUrl: 'http://mylisteningserver.com:9200/endpoint',
+    };
+
+    expect(() => {
+      validateConfig(actionType, config);
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"error validating action type config: error configuring send to Torq action: target url is not present in allowedHosts"`
+    );
+  });
+});
+
+describe('params validation', () => {
+  test('params validation passes when a valid body is provided', () => {
```

Comment:
is there any validation? can you add a failed test? (maybe empty body)

Reply:
Non-emptiness is validated in the UI, here it doesn't matter if the body is empty.